slasher/db/kv/spanner_new.go

```diff
@@ -45,10 +45,15 @@ func persistFlatSpanMapsOnEviction(db *Store) func(key interface{}, value interf
 // for slashing detection.
 // Returns span byte array, and error in case of db error.
 // returns empty byte array if no entry for this epoch exists in db.
-func (db *Store) EpochSpans(ctx context.Context, epoch uint64) (*types.EpochStore, error) {
+func (db *Store) EpochSpans(ctx context.Context, epoch uint64, fromCache bool) (*types.EpochStore, error) {
 	ctx, span := trace.StartSpan(ctx, "slasherDB.EpochSpans")
 	defer span.End()
 
+	if fromCache && db.flatSpanCache.Has(epoch) {
```

Comment:
Should we be getting from cache if we have it?

Reply:
yes